crates/simulate/src/stochastic/mod.rs

```diff
@@ -0,0 +1,120 @@
+//! Distribution sampling and price process generation and plotting.
+
+#![allow(dead_code)]
+
+use rand::Rng;
+use rand_distr::{
+    Beta as BetaDistr, Exp as ExpDistr, Gamma as GammaDistr, Normal as NormalDistr,
+    Poisson as PoissonDistr,
+};
+
+pub mod price_process;
+
+/// A trait for distribution sampling.
+pub trait Distribution<T> {
+    /// Sample from a distribution
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> T;
+}
+
+/// Normal distribution parameters struct
+pub struct Normal {
+    mean: f64,
+    std_dev: f64,
+}
+
+impl Normal {
+    /// Create a new normal distribution
+    fn new(mean: f64, std_dev: f64) -> Normal {
+        Normal { mean, std_dev }
+    }
+}
+
+impl Distribution<f64> for Normal {
+    /// Sample from a normal distribution
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> f64 {
+        let normal = NormalDistr::new(self.mean, self.std_dev).unwrap();
+        rng.sample(normal)
+    }
+}
+
+/// Poisson distribution parameters struct
+pub struct Poisson {
+    mean: f64,
+}
+
+impl Poisson {
+    /// Create a new poisson distribution
+    fn new(mean: f64) -> Poisson {
+        Poisson { mean }
+    }
+}
+
+impl Distribution<f64> for Poisson {
+    /// Sample from a poisson distribution
+    fn sample<R: Rng + ?Sized>(&self, rng: &mut R) -> f64 {
+        let poisson = PoissonDistr::new(self.mean).unwrap();
```

Comment:
This might not be a good idea. You're going to make a new distribution each time you want to sample? I think we could just have a distribution type enum so that we can select distributions and also avoid recreating a distribution each time we want to sample. What do you think?

Reply:
Also if you already have a distribution created in `new()` for `impl Poisson` (for example), then you can just do `self.sample()` basically